lib/system.nim

```diff
@@ -1629,7 +1629,7 @@ else:
       result[i+1] = y[i]
 
 
-proc astToStr*[T](x: T): string {.magic: "AstToStr", noSideEffect.}
+proc astToStr*(x: untyped): string {.magic: "AstToStr", noSideEffect.}
```

Comment:
This is a breaking change though, for example, now ``astToStr(2 + 3)`` produces something different. Arguably something better, but ... breaking change!

Reply:
not sure what you mean; both before and after PR I get:
```nim
echo 'echo astToStr(2+3)' | nim c -r -f -
2 + 3
```
(where the -f is needed until https://github.com/nim-lang/Nim/pull/13506 gets merged)